pkg/netceptor/packetconn.go

```diff
@@ -41,24 +60,24 @@ type PacketConn struct {
 	cancel          context.CancelFunc
 }
 
-func NewPacketConnWithConst(s *Netceptor, service string, advertise bool, adtags map[string]string, connTypeDatagram byte) *PacketConn {
+func NewPacketConnWithConst(s NetcForPacketConn, service string, advertise bool, adtags map[string]string, connTypeDatagram byte) PacketConner {
```

Comment:
Accept interfaces, return structs. Returning a `PacketConner` interface here forces anyone using this to use your interface and this is against the pattern we would like to use :smile: